code/modules/client/preferences.dm

```diff
@@ -547,6 +547,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				else
 					HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[NON-HUMAN\]</b></font></td></tr>"
 				continue
+			if((rank == "AI") && !(config.allow_ai))
+				HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[DISABLED\]</b></font></td></tr>"
+				continue

```

Comment:
@Cyberboss you don't get it, the EXACT SAME HTML code is used for literally every other thing, BANNED, NON-HUMAN, etc. So the HTML exploit can be used on EVERYTHING ELSE. I need to know how to fix it on everything else. I do not simply need to know what to do about the AI slot.